lib/chef/policy_builder/policyfile.rb

```diff
@@ -103,7 +103,7 @@ def initialize(node_name, ohai_data, json_attribs, override_runlist, events)
         end
 
         if Chef::Config[:environment] && !Chef::Config[:environment].chomp.empty?
-          raise UnsupportedFeature, "Policyfile does not work with Chef Environments."
+          raise UnsupportedFeature, "Policyfile does not work with #{Chef::Dist::PRODUCT} Environments."
```

Comment:
Could maybe omit the reference to the distro name with a rewording.

```suggestion
          raise UnsupportedFeature, "Policyfile does not work with an Environment configured."
```

Beyond the scope of minimal changes in this PR, but in the future I'd consider going further in the error message with something like `"Environment configured: #{Chef::Config[:environment]}. Policyfiles and Environments are mutually exclusive. You must use one or the other."`